### PR TITLE
provision script

### PR DESCRIPTION
**Pull request by AntonStasheuski**
# Original pull request
https://github.com/AntonStashevski/DevOps_TmS/pull/3
# Task

Create provisioning script for vagrant(Centos 8.3). Instance should have static IP for SSH connection.

1. Provision script(hw8_vagrant_provision.sh) should do following actions:
    * Create service user hw8. Home folder = /opt/hw8. Add ssh public key for           this user.
    * Add this user to sudoers group(sudo without password)
    * Update /etc/hosts file with record “127.0.0.1 myownapp.com”
    * Restrict ssh with access via password
    * Restrict ssh root access
    * Install mc, vim and git packages
    * Update DNS to 8.8.8.8 and 1.1.1.1
    * Check that address myownapp.com can be resolved. If not execute step 3
    * Print external(White) IP of you vagrant centos instance
    * Create dir /var/log/myownapp and set up hw8 user as owner of the dir
2. Learn rsync command
3. Refresh your knowledge regarding useful bash commands and 12 factors
4. How to put iftop command output to the file?

# Description

In general terms figured out the work with Vagrant. Write provision script and vagrant config.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Images

Screenshot 2021-02-21 at 15 58 44

Screenshot 2021-02-24 at 19 35 07

### CODE REVIEW

File: vagrant_bash_3/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  # config.vm.network "forwarded_port", guest: 80, host: 8080, host_ip: "127.0.0.1"
+
+  # Create a private network, which allows host-only access to the machine
+  # using a specific IP.
+  # config.vm.network "private_network", ip: "192.168.33.10"
+
+  # Create a public network, which generally matched to bridged network.
+  # Bridged networks make the machine appear as another physical device on
+  # your network.
+  # config.vm.network "public_network"
```
Comment (andreivaloshyn) [inline vagrant_bash_3/Vagrantfile:68]: instance should have custom IP

File: vagrant_bash_3/hw9_vagrant_provision.sh [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  echo "Start create user: $USER_NAME"
+  sudo useradd -s /bin/bash -d ${HOME_DIR}/${USER_NAME} -m -G wheel ${USER_NAME}
+}
+
+function ssh_key {
+  echo "Generate ssh keys for: $USER_NAME"
+  sudo mkdir ${HOME_DIR}/${USER_NAME}/.ssh
+  sudo chmod 700 ${HOME_DIR}/${USER_NAME}/.ssh
+  sudo su $USER_NAME
+  ssh-keygen -t rsa -f ${HOME_DIR}/${USER_NAME}/.ssh/id_rsa -q -P "" -C "${USER_NAME}@${VM_NAME}"
```
Comment (andreivaloshyn) [inline vagrant_bash_3/hw9_vagrant_provision.sh]: should be an existing key to be able to log in.

File: vagrant_bash_3/Vagrantfile (hunk 1 above)
Comment (AntonStasheuski) [reply vagrant_bash_3/Vagrantfile:68]: set static ip

File: vagrant_bash_3/hw9_vagrant_provision.sh (hunk 2 above)
Comment (AntonStasheuski) [reply vagrant_bash_3/hw9_vagrant_provision.sh]: use existing key

**Review by andreivaloshyn — APPROVED** [review]
Looks very good! Well done!

**Pull request merged**